Simbody/src/CableSpan.cpp

```diff
@@ -0,0 +1,3028 @@
+                tangent_S,
+                length,
+                initIntegratorStepSize,
+                tols.geodesicIntegratorAccuracy,
+                tols.constraintProjectionTolerance,
+                tols.constraintProjectionMaxIterations,
+                [&](const ContactGeometry::GeodesicKnotPoint& q)
+                {
+                    // Store the knot points in the cache.
+                    dataInst.geodesicIntegratorStates.push_back(q);
+                });
+
+            // Copy the state at the boundary frames.
+            geodesicBoundaryStates = {
+                dataInst.geodesicIntegratorStates.front(),
+                dataInst.geodesicIntegratorStates.back(),
+            };
+        }
+
+        // Use the geodeis state at the boundary frames to compute the fields in
+        // CurveSegmentData::Instance.
+
+        // Compute the Frenet frames at the boundary frames.
+        dataInst.X_SP = calcFrenetFrameFromGeodesicState(
+            geometry,
+            geodesicBoundaryStates.front());
+        dataInst.X_SQ = calcFrenetFrameFromGeodesicState(
+            geometry,
+            geodesicBoundaryStates.back());
+
+        // Compute the jacobi field scalars at the final frame.
+        dataInst.jacobi_Q = {
+            geodesicBoundaryStates.back().jacobiTrans,
+            geodesicBoundaryStates.back().jacobiRot,
+        };
+        dataInst.jacobiDot_Q = {
+            geodesicBoundaryStates.back().jacobiTransDot,
+            geodesicBoundaryStates.back().jacobiRotDot,
+        };
+
+        // Compute the curvatures at the boundary frames.
+        dataInst.curvatures_P = {
+            calcSurfaceCurvature(geometry, dataInst.X_SP, TangentAxis),
+            calcSurfaceCurvature(geometry, dataInst.X_SP, BinormalAxis),
+        };
+        dataInst.curvatures_Q = {
+            calcSurfaceCurvature(geometry, dataInst.X_SQ, TangentAxis),
+            calcSurfaceCurvature(geometry, dataInst.X_SQ, BinormalAxis),
+        };
+
+        // Compute the geodesic torsion at the boundary frames.
+        dataInst.torsion_P = geometry.calcSurfaceTorsionInDirection(
+            dataInst.X_SP.p(),
+            getTangent(dataInst.X_SP));
+        dataInst.torsion_Q = geometry.calcSurfaceTorsionInDirection(
+            dataInst.X_SQ.p(),
+            getTangent(dataInst.X_SQ));
+
+        // Store the arc length.
+        dataInst.arcLength = length;
+
+        // Update the initial integrator step size:
+        if (!useAnalyticGeodesic) { // does not apply to analytic surfaces.
+
+            // We want to know if we should attempt a larger initial step size
+            // next time, or a smaller one. If the initIntegratorStepSize was
+            // rejected, it makes little sense to start with the exact same step
+            // size next time, only to find it rejected again.
+
+            const int numStepsTaken =
+                dataInst.geodesicIntegratorStates.size() - 1;
```

Comment:
Assert `dataInst.geodesicIntegratorStates.size() > 0`?

Reply:
fixed by casting `size_t` to `int` before use